src/simulator/src/data.ts

```diff
@@ -0,0 +1,83 @@
+import { fullView } from './ux'
+import { createSubCircuitPrompt } from './subcircuit'
+import save from './data/save'
+import load from './data/load'
+import createSaveAsImgPrompt from './data/saveImage'
+import {
+    clearProject,
+    newProject,
+    saveOffline,
+    openOffline,
+    recoverProject,
+} from './data/project'
+import { createNewCircuitScope } from './circuit'
+import { createCombinationalAnalysisPrompt } from './combinationalAnalysis'
+import { colorThemes } from './themer/themer'
+import { showTourGuide } from './tutorials'
+import {
+    createVerilogCircuit,
+    // saveVerilogCode,
+    // resetVerilogCode,
+    // applyVerilogTheme,
+} from './Verilog2CV'
+import { generateVerilog } from './verilog'
+import { bitConverterDialog } from './utils'
+import { keyBinder } from '#/components/DialogBox/CustomShortcut.vue'
+import { ExportProject } from '#/components/DialogBox/ExportProject.vue'
+import { ImportProject } from '#/components/DialogBox/ImportProject.vue'
+
+const logixFunction = {
+    save,
+    load,
+    createSaveAsImgPrompt,
+    clearProject,
+    newProject,
+    saveOffline,
+    createOpenLocalPrompt: openOffline,
+    recoverProject,
+    createSubCircuitPrompt,
+    createCombinationalAnalysisPrompt,
+    fullViewOption: fullView,
+    colorThemes,
+    showTourGuide: showTourGuideHelper,
+    newVerilogModule: createVerilogCircuit,
+    generateVerilog,
+    bitconverter: bitConverterDialog,
+    createNewCircuitScope,
+    customShortcut: keyBinder,
+    ExportProject,
+    ImportProject,
+}
+// logixFunction.save = save
+// logixFunction.load = load
+// logixFunction.createSaveAsImgPrompt = createSaveAsImgPrompt
+// logixFunction.clearProject = clearProject
+// logixFunction.newProject = newProject
+// logixFunction.saveOffline = saveOffline
+// // logixFunction.newCircuit = newCircuit
+// logixFunction.createOpenLocalPrompt = openOffline
+// logixFunction.recoverProject = recoverProject
+// logixFunction.createSubCircuitPrompt = createSubCircuitPrompt
+// logixFunction.createCombinationalAnalysisPrompt =
+//     createCombinationalAnalysisPrompt
+// logixFunction.fullViewOption = fullView
+// logixFunction.colorThemes = colorThemes
+// logixFunction.showTourGuide = showTourGuideHelper
+// logixFunction.newVerilogModule = createVerilogCircuit
+// // logixFunction.saveVerilogCode = saveVerilogCode
+// // logixFunction.resetVerilogCode = resetVerilogCode
+// logixFunction.generateVerilog = generateVerilog
+// // logixFunction.applyVerilogTheme = applyVerilogTheme
+// logixFunction.bitconverter = bitConverterDialog
+// logixFunction.createNewCircuitScope = createNewCircuitScope
+// logixFunction.customShortcut = keyBinder
+// logixFunction.ExportProject = ExportProject
+// logixFunction.ImportProject = ImportProject
+export default logixFunction
+
+// Hack to restart tour guide
+function showTourGuideHelper() {
```

Comment:
let's use templating for any new function or variables, this is how we will expand the coverage of typescript. Atleast we should force it in new declaration